deltacat/compute/compactor/utils/primary_key_index.py

```diff
@@ -209,7 +208,8 @@ def group_hash_bucket_indices(
         if obj is not None:
             obj_ref = ray.put(obj)
             object_refs.append(obj_ref)
-            hash_bucket_group_to_obj_id[hb_group] = cloudpickle.dumps(obj_ref)
+            hash_bucket_group_to_obj_id[hb_group] = obj_ref
+            del obj_ref
```

Comment:
Though this approach is cleaning up object store, its not cleaning up main memory.

Reply:
I've experimented with the free internal API but it's not having any impact on cleaning up main memory.

Reply:
Running  `ray globalgc` has no impact on the main memory usage as well.

Reply:
Makes sense. Let's revisit if it becomes a blocker.